src/new_index/schema.rs

```diff
@@ -593,12 +596,17 @@ impl ChainQuery {
         limit: usize,
     ) -> Result<(UtxoMap, Option<BlockHash>, usize)> {
         let _timer = self.start_timer("utxo_delta");
+        let start = Instant::now();
         let history_iter = self
             .history_iter_scan(b'H', scripthash, start_height)
             .map(TxHistoryRow::from_row)
-            .filter_map(|history| {
-                self.tx_confirming_block(&history.get_txid())
-                    .map(|b| (history, b))
+            .map(|history| {
+                let height = history.key.confirmed_height;
+                (
+                    history,
+                    self.blockid_by_height(height as usize)
```

Comment:
it seems the original behavior will filter out utxos from orphaned blocks: https://github.com/Blockstream/electrs/blob/new-index/src/new_index/schema.rs#L894-L895, but the new behavior might not

do you know how the new behavior handles orphaned blocks?

Reply:
@mi-yu you're right about this, so just modified the code the use Solution 2: 

1. change the tx history schema so that we will save `blockhash` for that tx in the history db
2. when calculation delta, we first seek the `blockhash` for the history tx and see if this `blockhash` points an orphaned block, if that, we will filter out this tx for utxo/stats calculation
3. if `blockhash` not found for that history tx, fallback to old logic to use db to try to find the confirmed block(maybe orphaned too, code logic already taken care by old codes)
4. provided a backfill script to backfill `blockhash` for all history tx data which may take the same time to do index from scratch at the first time electrs start up. (~1.5 hours according to electrs official doc)